CDCgov/prime-reportstream
# Stable Smoke Tests in Github Actions

This PR stabilizes the smoke tests in our CI.

Before this the `volume` used by the postgres docker service could be manipulated by other steps or running jobs that caused smoke test failures. This removes the volume for CI runs.

Test Steps:
1. See that the Build Router check has passed, and the the Smoke Test step within that has passed.
2. Feel free to rerun the workflow to see it pass

Another method:
1. choose multiple passing commits from this branch
2. click rerun all jobs on for the Build Router workflow
3. see that they all pass and don't influence eachother
e.g.: 
- https://github.com/CDCgov/prime-reportstream/actions/runs/1472235578
- https://github.com/CDCgov/prime-reportstream/actions/runs/1467101999
- https://github.com/CDCgov/prime-reportstream/actions/runs/1463281590

Any other suggestions to assert that it won't cause the same trouble as before are greatly appreciated.

## Changes
- Use java setup action with caching
- Use gradle wrapper check to assert it's a valid jar
- Explicitly set up a local test environment for the CI run
- Add a postgres docker-compose with no persistent volume (I think this is what caused the earlier behavior)

## Checklist

## Fixes
- #2581 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- https://github.com/CDCgov/prime-reportstream/pull/3098
- https://github.com/CDCgov/prime-reportstream/issues/3104
